### DISCUSSION TITLE
Accelerating Sliders

### DISCUSSION THREAD

**Discussion by myutin** [ppy/osu#27264] [Ideas]
Note: I will be referring to "accelerating sliders" a lot in this post. To save me time typing, and save everyone else time reading, I will shorten this to "accelers".

Note: I know there are ways besides Sliderator for sliders to have non-linear movement (e.g. tumors, zigzags). Some might call these also "accelerating sliders". However, in this conversation, I am using "acceler" strictly to refer to a slider whose effective velocity is not constant along its length, but also is not visually distinct from a conventional slider.

Note: This question falls somewhere between Mapping Techniques and Development. Let me know if I should repost the discussion elsewhere. (This is already a repost of [a forum post](https://osu.ppy.sh/community/forums/topics/1885878?n=1))

As far as I understand, there is demand for accelers to be rankable. As far as I understand, the only practical way to make accelers is the Sliderator tool in Mapping Tools. However, sliders made with Sliderator are unrankable. I don't contest this particular decision --- the abundance of control points by Sliderator causes crashes in Lazer. The result of this, however, is effectively that no ranked maps can use accelers.

As far as I understand, the current ranking criteria do not explicitly disallow the concept of an acceler --- the criteria were written before the advent of Sliderator. My personal opinion is that accelers can be made readable with sufficient effort by the mapper. If future ranking criteria were to explicitly mention accelers, in my opinion, they should use the same level of scrutiny as is used for frequent SV changes. But I am not here to discuss potential revisions to the ranking criteria. I mention them just to say: the only reason all existing accelers are unrankable is the practical consideration in Lazer.

In my imagination, some future feature could conceivably allow the creation of accelers which do not crash Lazer. Since Stable is feature-locked, such a feature would likely not result in these accelers functioning on Stable. They could, of course, be converted to conventional sliders when played in Stable, but they would nevertheless function differently on Stable versus Lazer. My first question is: how acceptable is it to add mapping features to Lazer that do not carry over in Stable? I know there is already precedent for gameplay changes (e.g. sliderhead accuracy), but I do not know how far this precedent can stretch.

My second question is: can someone point me to the most up-to-date / most detailed guidelines or discussions on etiquette for contributing to feature development? If someone (e.g. me) were to develop a feature/PR to allow creation of accelers in the Lazer editor, what would it take for the feature to be approved?

Lastly, are there any possible problems I may have missed? Are there any reasons why accelers should not exist in osu? Are there any reasons why such a feature should not be developed?

Wishing everyone a good day!

**Comment by bdach** [comment]
We know people want this but we do not want "sliderator" to be used because it abuses control point count, causing performance issues among other things.

Yes we'll likely want to support this but lazer has to be at feature parity with stable first and we still have work to do in that respect (editor, storyboarding, etc.)  I don't think any new object types are happening before that.
